### ISSUE TITLE
Count is wrong after hoisting left side up.

### ISSUE THREAD

**Issue by gnachman** [aroemers/objc-aatree#1]
```
    AATree *tree = [[AATree alloc] initWithKeyComparator:^NSComparisonResult(id  _Nonnull obj1, id  _Nonnull obj2) {
        return [obj1 compare:obj2];
    }];
    tree[@243] = @"_243";
    tree[@162] = @"_162";
    tree[@324] = @"_324";
   
    assert(tree.count == 3);  // passess
    [tree removeObjectForKey:@243];
    assert(tree.count == 2);  // fails
```

I know this project is long-dead, so I'm making this issue to help anyone in the future who runs in to this problem. A solution will be forthcoming in the iTerm2 code.

**Comment by aroemers** [comment]
Hi @gnachman, thank you for the report. If I can merge some fixes or anything, let me know.

**Comment by aroemers** [comment]
Fixed by #2.
